### DISCUSSION TITLE
Why are association tables generated as Table and not as Declarative Model?

### DISCUSSION THREAD

**Discussion by dirkdevries96** [agronholm/sqlacodegen#174] [Q&A]
I'm interested in generating association table using Declarative Base. This way this table can be referenced elsewhere in a declarative way. 

But the readme states the following:
### Model class generators
The code generators that generate classes try to generate model classes whenever possible. There are two circumstances in which a Table is generated instead:
- the table has no primary key constraint (which is required by SQLAlchemy for every model class)
- the table is an association table between two other tables (see below for the specifics)

Is there a specific reason why it's done this way? Is perhaps my use case of referencing an association table not good practice?

Below is an expample of the current code generated.

```
class ProfileType(Base):
    __tablename__ = 'profile_type'

    id = Column(Integer, primary_key=True, server_default=text("nextval('profile_type_id_seq'::regclass)"))
    name = Column(String, unique=True)

    project = relationship('Project', secondary='project_profile_type', back_populates='profile_type')

class Project(Base):
    __tablename__ = 'project'

    id = Column(Integer, primary_key=True, server_default=text("nextval('project_id_seq'::regclass)"))
    name = Column(String)

    profile_type = relationship('ProfileType', secondary='project_profile_type', back_populates='project')


t_project_profile_type = Table(
    'project_profile_type', metadata,
    Column('project_id', ForeignKey('project.id'), primary_key=True, nullable=False),
    Column('profile_type_id', ForeignKey('profile_type.id'), primary_key=True, nullable=False),
    Index('project_profile_type_project_id_profile_type_id_idx', 'project_id', 'profile_type_id', unique=True)
)
```

**Comment by agronholm** [comment]
The reason is that SQLAlchemy cannot keep such "association objects", in sync with the relationship attributes, causing some very puzzling bugs. If your intent is to not have a direct many-to-many relationship between Project and ProfileType, that's fine then, but sqlacodegen does not have a way for the user to communicate such intent at the moment, so you'll have to edit the generated code a bit.
